### ISSUE TITLE
Delete account functionality

### ISSUE THREAD

**Issue by liza-veis** [BinaryStudioAcademy/bsa-2023-calmpal#168]
- [ ] implement delete account functinality
- [ ] Add modal with steps before user deletes an account
- [ ] After deleting an account, user should be redirected to Sign In (we will change this later)

Note: we don't have design for now, so just try to stick to application style. Also we need to add modal in this issue #160, so you can cooperate with @Dmytro-Kucherenko 

<img width="670" alt="Screenshot 2023-09-03 at 16 48 10" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/64521658/574bb8d4-4166-46f4-928f-ce3a88183111">
<img width="249" alt="Screenshot 2023-09-03 at 16 02 08" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/64521658/5f81970c-f78c-45b3-a57f-792f980314b8">

**Comment by juanc-escobar** [comment]
Hi, @Liza-Veis I have one question, What should we do with the data from the form?

**Comment by liza-veis** [comment]
let's not do anything with this for now

**Comment by juanc-escobar** [comment]
Hi, @Liza-Veis One more question: Do I need to implement the backend functionality for deleting the account, or is that another PR ?  

also for this task do I need to wait for "My profile" and the "delete account" button to be implemented? since that is another PR. 

I am currently testing the modal with a mockup button:
![modal test](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/107894952/04ad30ef-a276-4855-993c-852d638c0ac9)

**Comment by juanc-escobar** [comment]
can we use custom Hooks?

**Comment by juanc-escobar** [comment]
Hi, @Liza-Veis I was talking with @Dmytro-Kucherenko  about the modal component since multiple developers will need to use it. He already finished the modal with styles (looks good), I was working on the modal too, is not finished yet but want to know if I should use the @Dmytro-Kucherenko modal,  or keep working on my own thing until his code is merged and then refactor to his modal?

**Comment by dmytro-kucherenko** [comment]
We defo need to create one common modal component, only the question which will be correctly implemented)

**Comment by juanc-escobar** [comment]
@Dmytro-Kucherenko  Oh, I see, so I'm going to try to implement my functionality on the component and see if we need to implement any additional changes to receive multiple steps in the modal but also handle a single step.

**Comment by juanc-escobar** [comment]
Hi, @Liza-Veis one question this task depends on **cp-61: profile settings sidebar** If I manage to finish before cp-61 is merged, what should I do? Should I wait for cp-61 to be merged or do a PR with a mockup button meanwhile?

**Comment by liza-veis** [comment]
Hi! Create a mockup button for now, open a draft pr and write in the description that it depends on another pr

**Comment by juanc-escobar** [comment]
Hi, @Liza-Veis, @fshabanov, @v0ldemar01 could you review this draft while the profile settings get merged?

Let  me know if it is posible.

Thanks.

**Comment by KrychkovskaVira** [comment]
1. There are no notifications for required fields if no options are selected, although the hand cursor is present.
2. The user can also proceed further by selecting "other" without entering anything in the input field.
3. After closing the window, the previous selected and "Delete Account" tabs are highlighted



https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/88768615/059a8854-12db-4979-9b77-e5c6b1f499c3
